proposals/xxxx-die.md

```diff
@@ -0,0 +1,215 @@
+# Introducing Namespacing for Common Swift Error Scenarios
+
+* Proposal: [SE-TBD](0tbd-fatal.md)
+* Authors: [Dave DeLong](http://github.com/davedelong), [Tim Vermeulen](https://github.com/timvermeulen), [Erica Sadun](https://github.com/erica)
+* Review Manager: TBD
+<!---
+* Implementation: [apple/swift#NNNNN](https://github.com/apple/swift/pull/NNNNN)
+* Decision Notes: [Rationale](https://lists.swift.org/pipermail/swift-evolution/), [Additional Commentary](https://lists.swift.org/pipermail/swift-evolution/)
+* Bugs: [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN), [SR-MMMM](https://bugs.swift.org/browse/SR-MMMM)
+* Previous Revision: [1](https://github.com/apple/swift-evolution/blob/...commit-ID.../proposals/NNNN-filename.md)
+* Previous Proposal: [SE-XXXX](XXXX-filename.md)
+* -->
+
+## Introduction
+
+This proposal provides a namespaced umbrella for common exit scenarios, modernizing a holdover from C-like languages.
+
+Discussion: [Introducing Namespacing for Common Swift Error Scenarios](https://forums.swift.org/t/introducing-namespacing-for-common-swift-error-scenarios/10773)
+
+## Motivation
+
+Swift's `fatalError` is a one-size-fits-all hammer. Swift lacks a modern extensible solution that differentiates fatal error scenarios for specific contexts. Because of this, many developer libraries now implement custom exit points like the following example to provide that functionality:
+
+```swift
+/// Handles unimplemented functionality with site-specific information
+/// courtesy of Soroush Khanlou, who discovered it from Caleb Davenport, 
+/// who discovered it from Brian King
+func unimplemented(_ function: String = #function, _ file: String = #file) -> Never {
+    fatalError("\(function) in \(file) has not been implemented")
+}
+```
+
+Namespacing these marker functions to a common type results in groupable, extensible, and discoverable IDE elements.
+
+There are four universal core language concepts that can be better represented in Swift. They are:
+
+* This code is unreachable.
+* This code is not yet implemented.
+* This method must be overriden in a subtype.
+* This code is should never be called.
```

Comment:
Typo: `This code should never be called.`

Reply:
Fixing